src/lib.rs

```diff
@@ -51,25 +51,43 @@ pub fn run(args: &clap::ArgMatches, current_dir: &PathBuf) -> Result<Vec<Warning
         };
 
         let mut lines = get_line_entries(&fe, strs);
+        let mut backup_file = None;
 
         // run fixers & write results to file
         let mut result = checks::run(&lines, &skip_checks);
         if is_fix && !result.is_empty() && fixes::run(&mut result, &mut lines, &skip_checks) > 0 {
             // create backup copy unless user specifies not to
             let should_backup = !args.is_present("no-backup");
             if should_backup {
-                let backup_file = fs_utils::backup_file(&fe)?;
-                println!("Original file was backed up to: {:?}\n", backup_file);
+                backup_file = Some(fs_utils::backup_file(&fe)?.into_os_string());
             }
 
             // write corrected file
             fs_utils::write_file(&fe.path, lines)?;
         }
 
-        warnings.extend(result);
+        let mode = if is_fix {
```

Comment:
I think we can do this once before the loop, when we read `--fix` argument.

Reply:
@Anthuang Can you make these changes?